azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1655,8 +1662,13 @@ private void ajaxHandleUpload(final HttpServletRequest req, final HttpServletRes
       props.put(ValidatorConfigs.CUSTOM_AUTO_FIX_FLAG_PARAM, "true");
     }
 
-    if (this.lockdownUploadProjects) {
-      registerError(ret, "Project uploading is locked out", resp, 400);
+    if (this.lockdownUploadProjects && !UserUtils
+        .hasPermissionforAction(this.userManager, user, Type.UPLOADPROJECTS)) {
+      final String message =
+          "Project uploading is locked out. Only admin users and users with special permissions can upload projects. "
+              + "User " + user.getUserId() + " doesn't have permission to upload project.";
+      logger.info(message);
+      registerError(ret, message, resp, 400);
```

Comment:
Is 403 response code more appropriate?
https://stackoverflow.com/questions/3297048/403-forbidden-vs-401-unauthorized-http-responses